.github/workflows/ci_linux_x64_clang_tsan.yml

```diff
@@ -37,21 +37,28 @@ jobs:
       - environment=${{ needs.setup.outputs.runner-env }}
       - cpu
       - os-family=Linux
+
+    container:
+      # TODO(scotttodd): use ghcr.io/iree-org/cpubuilder_ubuntu_jammy_x86_64 here
+      image: gcr.io/iree-oss/base-bleeding-edge@sha256:cf2e78194e64fd0166f4141317366261d7a62432b72e9a324cb8c2ff4e1a515a
+      # TSan in particular needs some settings that this option includes:
+      #   * https://github.com/google/sanitizers/issues/1716
+      #   * https://security.stackexchange.com/q/214923
+      options: --privileged
```

Comment:
Since the container is based on Ubuntu 22.04 + clang >= 18.1.0 is used, setting `vm.mmap_rnd_bits` shouldn't not be necessary. Hence, I would like to better understand what the `--privileged` option is needed for. I am not feeling super comfortable with it as the host isn't shielded any longer from the container.

Reply:
I can run more experiments and respond in more detail later, but note that the existing scripts, which this is migrating away from, already set `--privileged` _for all docker runs_: https://github.com/iree-org/iree/blob/0d196cd86e735d4afdd7a6db40d1228dbaa8f779/build_tools/docker/docker_run.sh#L111-L116

This is changing to only keep that for TSan.

Reply:
In that case what you did only makes things better 😆 Therefore, maybe let's fill an issue in order to not forget that we need to re-evaluate if we really need `--privileged` and let's just land this!

Reply:
Added a note about not relying on Docker for sandboxing.